srtcore/queue.cpp

```diff
@@ -1529,15 +1529,15 @@ void CRcvQueue::stopWorker()
     m_bClosing = true;
 
     // Sanity check of the function's affinity.
-    if (pthread_equal(pthread_self(), m_WorkerThread))
-    {
-        LOGC(mglog.Error, log << "IPE: RcvQ:WORKER TRIES TO CLOSE ITSELF!");
-        return; // do nothing else, this would cause a hangup or crash.
-    }
+    //if (pthread_equal(pthread_self(), m_WorkerThread))
```

Comment:
What's wrong with replacing it with
```
if (this_thread::get_id() == m_WorkerThread.get_id())
```
You'll just have to implement `this_thread` thing for POSIX.

Reply:
Created a dedicated function.
```c++
bool CheckIfThisThread(const CThread& th);
```